### PR TITLE
feat: add counter and caption via a state manager

### PR DESCRIPTION
**Pull request by varholak-peter**
https://jira.economist.com/browse/WP-415

### CODE REVIEW

File: src/example.js [hunk 1]
```diff
@@ -1,33 +1,57 @@
...
+import CarouselStateManager from './parts/carousel-state-manager';
+import CarouselCounter from './parts/carousel-counter';
+import CarouselCaptionDisplay from './parts/carousel-caption-display';
+import CarouselCreditDisplay from './parts/carousel-credit-display';
 
 export const horizontalNodes = ([
-  <div className="example-node" key="0">
+  <div className="example-node"
+    key="0"
+    data-caption="This i the caption 1"
```
Comment (yasgreen93) [inline src/example.js]: Is this a typo?

File: src/example.js [hunk 2]
```diff
@@ -1,33 +1,57 @@
...
+    data-credit="Credit for 1"
+  >
     <div className="example-node__image" />
     <h3 className="example-node__title">Work for us: Nico Colchester journalism fellowshiphs</h3>
     <h4 className="example-node__subtitle">Europe | 1 hour 44 mins ago</h4>
   </div>,
-  <div className="example-node" key="1">
+  <div className="example-node"
+    key="1"
+    data-caption="This i the caption 2"
```
Comment (yasgreen93) [inline src/example.js]: Same here and the rest below.

File: src/index.js [hunk 3]
```diff
@@ -57,6 +58,9 @@ export default class Carousel extends React.Component {
       if (typeof this.props.onScrollerCreated === 'function') {
         this.props.onScrollerCreated(this.scroller);
       }
+      if (typeof this.props.onScrollerSegmentdidchange === 'function') {
```
Comment (yasgreen93) [inline src/index.js]: Could this be `onScrollerSegmentDidChange` ?

Comment (varholak-peter) [inline src/index.js]: It is a name they have in https://github.com/ftlabs/ftscroller API so won't change for transparency reasons.

**Pull request merged**